### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by zachcran**
**Is this pull request associated with an issue(s)?**
No.

**Description**
This PR fixes the issues left over from #78 after it was merged.

**TODOs**
- [x] Module documentation
- [x] Remove parameter prefixes in return value documentation
- [x] Remove reference to missing label

### CODE REVIEW

File: cmake/cmakepp_lang/map/map.cmake [hunk 1]
```diff
@@ -1,3 +1,8 @@
+#[[[ @module
+# Provides an easy way for the user to include the various functions used 
+# to create and maintain an instance of a CMakePPLang map.
+#]]
```
Comment (ryanmrichard) [inline cmake/cmakepp_lang/map/map.cmake:4]: Can you put the doc comments for the module under the license? Technically I think they aren't licensed if you put them above it.

Comment (zachcran) [reply cmake/cmakepp_lang/map/map.cmake:4]: Oh yes, I misread the documentation and thought they had to be at the very top of the file. Comments can be above them.

File: cmake/cmakepp_lang/serialization/detail/serialize_list.cmake
Comment (zachcran) [inline cmake/cmakepp_lang/serialization/detail/serialize_list.cmake:1]: @ryanmrichard Should your old licensing headers be removed from these files in `cmake/cmakepp_lang/serialization/detail`?

Comment (ryanmrichard) [reply cmake/cmakepp_lang/serialization/detail/serialize_list.cmake:1]: Yeah. The org owns all of the code (and is also liable should it ever break stuff)

**Review by ryanmrichard — APPROVED** [review]

**Comment by zachcran** [comment]
@AutonomicPerfectionist r2g

**Pull request merged**